### ISSUE TITLE
[Feature] Use desktop version of Google search

### ISSUE THREAD

**Issue by olivierobert** [paladin3895/google-search#3]
## Issue

While using specific user agents is expected, the current implementation of the scraper seems to use the mobile version of the Google search page (not the header with the 4 tabs)

![image](https://user-images.githubusercontent.com/696529/212828315-2b454ae6-a318-4d0e-a730-bd9751d8bde4.png)

As a result, the number of links and ads is quite different from the desktop version:

|Mobile|Desktop|
|-|-|
|![image](https://user-images.githubusercontent.com/696529/212829025-507a68f5-dcaf-40c2-9e2a-42070b18e684.png)|![image](https://user-images.githubusercontent.com/696529/212828915-67259fce-5272-4bd8-a62f-6b1e1070be07.png)|

## Expected

The scraper should use the desktop version of the Google search page:

![image](https://user-images.githubusercontent.com/696529/212828632-4230e69c-dcc4-4847-8375-fa992a2e3448.png)

This part of the code should be amended:

https://github.com/paladin3895/google-search/blob/9e652a7fab9f1073318c4bbb09de4fafba9de5cc/app/Utils/Search/GoogleSearch.php#L37-L42

Note that I have also not seen any implementation for the rotation of user agents:

![image](https://user-images.githubusercontent.com/696529/212829410-969814eb-f0ef-415f-afab-e83f0c7629d7.png)

**Comment by paladin3895** [comment]
Hi Olivier,

The search page HTML was a desktop version of browsers with javascript option turned off (user-agent was set as Internet Explorer 9 in this instance). Because when searching using Chrome user-agent, Google returned an HTML page with no search results but the results will be rendered later by javascript that makes it very hard to scrape the data from the page. So one workaround I made was to send the query as an old browser to disable javascript and get the results in raw HTML.

I have checked the results thoroughly and see that the search results between these 2 versions were identical. Please note that this query `document.querySelectorAll('a').length` would return all the `a` tags that were not related to the search results (some were UI components like buttons, previous search links, feedback links, etc)

Hope this explains it.

**Comment by paladin3895** [comment]
Your other question about the IP addresses and user-agent rotation was noted. I'll update the source code and the demo server along with other improvements tomorrow.

**Comment by olivierobert** [comment]
Well noted on the issue you faced with JS but there might then be an issue with the Guzzle HTTP client. You could try this user agent `"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.0.0 Safari/537.36"` as they have been tried and validated to be working:

As for the querying all the links, it is indeed expected from the requirements (so not only search result links):

<img width="601" alt="image" src="https://user-images.githubusercontent.com/696529/212915149-64c54ba8-34f5-4ccf-8042-4c96262aba83.png">

**Comment by paladin3895** [comment]
Indeed, there were discrepancies between javascript and non-javascript versions of the search page. I just found a solution to this problem: basically we can get the search page with javascript and then run a headless Chrome to render HTML for us to scrape the data.
I'll push a feature branch tomorrow morning to address this.

**Comment by paladin3895** [comment]
Please refer to this commit #[4014](https://github.com/paladin3895/google-search/pull/9/commits/401437becfde08c76de7a02234d0077ebbd3b821)
- Make rotating proxy servers and user-agents as part of the search config.
- Fix the desktop version of Google search.
- Fix counting all the links instead of just search related links.
